drivers/crypto/Kconfig.stm32

```
@@ -3,8 +3,12 @@
 # Copyright (c) 2020 Markus Fuchs <markus.fuchs@de.sauter-bc.com>
 # SPDX-License-Identifier: Apache-2.0
 
+DT_COMPAT_ST_STM32_AES := st,stm32-aes
+DT_COMPAT_ST_STM32_CRYP := st,stm32-cryp
+
 menuconfig CRYPTO_STM32
 	bool "STM32 Cryptographic Accelerator driver"
+	default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_AES)) || $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_CRYP))
 	depends on SOC_FAMILY_STM32
```

Probably it would be better to move the dt_compat_enabled from `default` to `depends on`:
```suggestion
	depends on $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_AES)) || $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_CRYP))`
```
Otherwise the driver will also be built if the TinyCrypt or mbedTLS stubs are used and needs to be explicitly disabled.
While this doesn't cause any conflicts it would waste flash/ram. Please let me know which one you prefer.

---

@erwango - please comment here

---

@str4t0m - I'm going to approve and merge. As it stands, this is the worse scenario, if I understand correctly. Reducing flash / ram could be considered an optimization and can be deferred. If you would like to make this change, please create a separate issue, reference this PR, and provide some numbers for ram / flash footprints with and without the change.

Thanks again for getting this in!